docs/BLSSignatureChecker.md

```diff
@@ -1 +1,196 @@
+    uint32 referenceBlockNumber, 
+    NonSignerStakesAndSignature memory params
+)
+    public 
+    view
+    returns (QuorumStakeTotals memory, bytes32)
+
+struct NonSignerStakesAndSignature {
+    uint32[] nonSignerQuorumBitmapIndices;
+    BN254.G1Point[] nonSignerPubkeys;
+    BN254.G1Point[] quorumApks;
+    BN254.G2Point apkG2;
+    BN254.G1Point sigma;
+    uint32[] quorumApkIndices;
+    uint32[] totalStakeIndices;
+    uint32[][] nonSignerStakeIndices;
+}
+
+struct QuorumStakeTotals {
+    uint96[] signedStakeForQuorum;
+    uint96[] totalStakeForQuorum;
+}
+```
+
+The goal of this method is to allow an AVS to validate a BLS signature formed from the aggregate pubkey ("apk") of Operators registered in one or more quorums at some `referenceBlockNumber`.
+
+Some notes on method parameters:
+* `referenceBlockNumber` is the reason each registry contract keeps historical states: so that lookups can be performed on each registry's info at a particular block. This is important because Operators may sign some data on behalf of an AVS, then deregister from one or more of the AVS's quorums. Historical states allow signature validation to be performed against a "fixed point" in AVS/quorum history.
+* `quorumNumbers` is used to perform signature validation across one *or more* quorums. Also, Operators may be registered for more than one quorum - and for each quorum an Operator is registered for, that Operator's pubkey is included in that quorum's apk within the `BLSApkRegistry`. This means that, when calculating an apk across multiple `quorumNumbers`, Operators registered for more than one of these quorums will have their pubkey included more than once in the total apk.
+* `params` contains both a signature from all signing Operators, as well as several fields that identify registered, non-signing Operators. While non-signing Operators haven't contributed to the signature, but need to be accounted for because, as Operators registered for one or more signing quorums, their public keys are included in that quorum's apk. Essentially, in order to validate the signature, nonsigners' public keys need to be subtracted out from the total apk to derive the apk that actually signed the message.
+
+This method performs the following steps. Note that each step involves lookups of historical state from `referenceBlockNumber`, but the writing in this section will use the present tense because adding "at the `referenceBlockNumber`" everywhere gets confusing. Steps follow:
+1. Calculate a total apk for all nonsigners, multiplying each nonsigner's pubkey by the number of `quorumNumbers` they are registered for.
+2. Negate the calculated total apk to "remove" the nonsigners' public keys.
+3. For each quorum:
+    * Add the quorum's total apk to the total apk being calculated. Because the negation of nonsigner pubkeys are already in the total apk, this effectively subtracts nonsigner pubkeys from the total apk (leaving only signing pubkeys)
```

Comment:
likewise, I think "total apk" -> "signing apk" could help with clarity/precision.
So I would revise this line to something like
```
    * Add the quorum's signing apk to the total (signing) apk being calculated
```
at least if you make the above addition to clarify how this signing apk is calculated

Reply:
[20583ba](https://github.com/Layr-Labs/eigenlayer-middleware/pull/116/commits/20583bab55930680795024e599a70ec71a400325)